### DISCUSSION TITLE
Warm Start Questions

### DISCUSSION THREAD

**Discussion by tmacattank** [osqp/osqp#546] [General]
Hi,
I have some problem when using warmstart. I try to manually set warm start primal values by using "osqp_warm_start_x", but osqp randomly fails to have feasible solution(Actually, osqp works well before I try to warm start the problem). Should warm start variables in feasible sets? Is there any possible reason that OSQP may fail when it is warm started?

**Comment by goulart-paul** [comment]
It is not necessary to warm start the solver at a primal feasible solution.   In principle any warm start values for the primal variables (or dual variables for that matter) will still allow the solver the converge.

It depends what you mean by 'fails' though.   For example, if you could give a warm start vector with extremely large norm then you might see the solver fail to converge within the iteration limit.    You might also have a really unfavourable value of `rho` set (say from a previous solution with different matrix entries), in which case convergence could be very slow.

It's quite difficult to say what the problem is without a MWE though.

**Comment by sichaozhang1112** [comment]
@goulart-paul 
Hello Prof. Goulart,
I've encountered similar problem. I tried osqp to solve the same input data set for 20 times, two of them failed (solution diverge and unacceptable) due to primal infeasibility, although the input is the same...
And if i turn on warm start (if last solution is successful) to solve a sequence of data set, the situation will be worse, it seems the solution with warm start is more likely to diverge?
Maybe i get the settings wrong. I would appreciate it if you could clear my doubts.

**Comment by lemolemac** [comment]
I have using osqp-cpp warm start, and I found cpp warm start nevers while python-version warm start works perfect. I use the test code 'test_basic_qp_warm_start()' and the iter num after warm start is around 20, and not 1.
